src/shared/errors.ts

```diff
@@ -0,0 +1,18 @@
+import { SizeEstimate } from "../utils/content-size"
```

Comment:
we dont use this pattern anywhere else, and i don't want to litter the shared directory with unnecessary types

Reply:
This type is used in three of the files in this PR.  I would think it should be in shared at that point.  Otherwise, we're going to have a "shared" Type that isn't in the shared area.

Reply:
ok, I think we are likely to want to migrate toward it. Defined error types are very helpful and make tracing way easier. @saito-sv, @Garoth what are your thoughts?